Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -372,4 +373,131 @@ final class DocumentTests: XCTestCase {
+            expect(doc).to(equal(["double": 3.0, "int32": 20, "bool": true, "decimal": Decimal128("100")]))
+            return
+        }
+
+        // overwrite int64 with int64
+        let bigInt64 = Int64(Int32.max) + 1
+        doc["int64"] = bigInt64
+
+        let bigInt = Int(bigInt64) + 1
+        doc["int64"] = bigInt
+
+        // final values
+        expect(doc).to(equal(["double": 3.0, "int32": 20, "int64": bigInt, "bool": true, "decimal": Decimal128("100")]))
+    }
+
+    // test replacing some of the non-Overwritable types with values of their own types
+    func testNonOverwritable() throws {
+        var doc = DocumentTests.nonOverwritables
+
+        doc["string"] = "hi"
+        doc["nil"] = nil
+        let newDoc: Document = ["y": 1]
+        doc["doc"] = newDoc
+        doc["arr"] = [3, 4]
+        let oid = ObjectId()
+        doc["oid"] = oid
+
+        expect(doc).to(equal(["string": "hi", "nil": nil, "doc": newDoc, "arr": [3, 4] as [Int], "oid": oid]))
+    }
+
+    // test replacing both overwritable and nonoverwritable values with values of different types
+    func testReplaceValueWithNewType() throws {
+
+        var doc1 = DocumentTests.overwritables
+
+        doc1["double"] = Int(10)
+        doc1["int32"] = "hi"
+        doc1["bool"] = [1, 2, 3] as [Int]
+        doc1["decimal"] = 100
+
+        if XCTestCase.is32Bit {
+            expect(doc1).to(equal(["double": 10, "int32": "hi", "bool": [1, 2, 3] as [Int], "decimal": 100]))
+        } else {
+            doc1["int64"] = Decimal128("1.0")
+            expect(doc1).to(equal(["double": 10, "int32": "hi", "int64": Decimal128("1.0"), "bool": [1, 2, 3] as [Int], "decimal": 100]))
+        }
+       
+        var doc2 = DocumentTests.nonOverwritables
+
+        doc2["string"] = 1
+        doc2["nil"] = "hello"
+        doc2["doc"] = "hi"
+        doc2["arr"] = 5
+        doc2["oid"] = 25.5
+
+        expect(doc2).to(equal(["string": 1, "nil": "hello", "doc": "hi", "arr": 5, "oid": 25.5]))
+    }
+
+    // test setting both overwritable and nonoverwritable values to nil
+    func testReplaceValueWithNil() throws {
+        var doc1 = DocumentTests.overwritables
+
+        doc1["double"] = nil
+        doc1["int32"] = nil
+        doc1["bool"] = nil
+        doc1["decimal"] = nil
+
+        if XCTestCase.is32Bit {
+            expect(doc1).to(equal(["double": nil, "int32": nil, "bool": nil, "decimal": nil]))
+        } else {
+            doc1["int64"] = nil
+            expect(doc1).to(equal(["double": nil, "int32": nil, "int64": nil, "bool": nil, "decimal": nil]))
+        }
+        
+        var doc2 = DocumentTests.nonOverwritables
+
+        doc2["string"] = nil
+        doc2["nil"] = nil
+        doc2["doc"] = nil
+        doc2["arr"] = nil
+        doc2["oid"] = nil
+
+        expect(doc2).to(equal(["string": nil, "nil": nil, "doc": nil, "arr": nil, "oid": nil]))
+    }
+
+    // Test types where replacing them with an instance of their own type is a no-op
```

Comment:
If this is comparable to replacing Overwritable types, perhaps it can also assert that storage is not duplicated.

Reply:
👍 done